GitCommands/Git/GitRefName.cs

```diff
@@ -92,5 +92,11 @@ public static bool IsRemoteHead([NotNull] string refName)
         {
             return _remoteHeadRegex.IsMatch(refName);
         }
+
+        [Pure]
+        public static GitRef CreateBranchRef(GitModule module, string guid, string name)
```

Comment:
Could you please clarify why this method is needed and why it is placed in `GitRefName`?
`GitRefName.RefsHeadsPrefix` is public, so it could be referenced from `RepoObjectsTree.Nodes.Branches` where necessary.
Right now it creates unnecessary types coupling...

Reply:
It was in the GitRef.cs, @drewnoakes moved the other functions to this file and deleted the CreateBranchRef. I will revisit to see if it can be optimized.

Reply:
No other functions in `GitRefName` require `GitModule`. I feel like this should move to `GitModule` for now, and be relocated as part of #4643.

Reply:
Ignore that. Just noticed you've already moved it somewhere sensible 👍